core/include/core/mask_store.hpp

```diff
@@ -7,29 +7,49 @@
 #pragma once
 
 #include <type_traits>
+#include <utility>
+#include <vecmem/containers/device_vector.hpp>
+#include <vecmem/memory/memory_resource.hpp>
 
+#include "definitions/basic_types.hpp"
+#include "definitions/detray_qualifiers.hpp"
 #include "utils/enumerate.hpp"
 #include "utils/indexing.hpp"
 
 namespace detray {
 
 /** A mask store that provides the correct mask containers to client classes. */
-template <template <typename...> class tuple_type = dtuple,
-          template <typename...> class vector_type = dvector,
+template <template <typename...> class vector_type = dvector,
```

Comment:
I do not like this... :frowning:

Please keep the type templated on the tuple type. I'm absolutely sure that it can be done. As I wrote in my previous comment, we should be able to keep the `detray::tuple`, or rather `detray::dtuple` type functional with `thrust::tuple` as well.

Reply:
No problem